python_bindings/trainer_wrap.cpp

```diff
@@ -15,7 +15,7 @@ void init_trainer(py::module &m) {
     py::class_<Trainer, PyTrainer>(m, "Trainer")
         .def(py::init<>())
         .def_readwrite("data_set", &Trainer::data_set_)
-        .def("train", &Trainer::train, py::arg("num_epochs") = 1);
+        .def("train", &Trainer::train, py::arg("num_epochs") = 1, py::call_guard<py::gil_scoped_release>());
```

Comment:
was there a reason to add this? seems like a critical change thats part of a misc edits PR?

Reply:
If this isn't added then the README example will not work and Pytorch throws an error when trainer.train() is called. 

Error:
```
RuntimeError: The autograd engine was called while holding the GIL. 
If you are using the C++ API, the autograd engine is an expensive operation that does not require the GIL to be held so you should release it with 'pybind11::gil_scoped_release no_gil;'. 
If you are not using the C++ API, please report a bug to the pytorch team.
```
Anders and I are aware of these issues and a future PR which adds the custom models will add py::call_guard<py::gil_scoped_release>() to the necessary calls. Similar to how the bindings are structured in this repo: https://github.com/NVIDIA/MinkowskiEngine/blob/master/pybind/extern.hpp#L523